### DISCUSSION TITLE
Timing points are cleared from beatmaps when parsed from JSON

### DISCUSSION THREAD

**Discussion by khang06** [ppy/osu#15327] [Q&A] [answered]
i'm not really sure if this is worth fixing since it doesn't look like beatmaps parsed from json are actually used anywhere right now, but it was affecting some stuff i was doing with them. the timing points are serialized fine, but don't get deserialized at all.

i suspect that this is because the fields that contain them in `ControlPointInfo` inherit `IReadOnlyList`, so they can't be deserialized properly

**Comment by smoogipoo** [comment, marked as answer]
Thanks for the report. https://github.com/ppy/osu/issues/15328
